File: Backend/EmployeeManagementSystem/EmployeeManagementSystem.Domain/Entities/Attendance.cs


using EmployeeManagementSystem.Domain.Enum;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace EmployeeManagementSystem.Domain.Entities
{
    public class Attendance
    {
        public int Id { get; set; }
        public int EmployeeId { get; set; }
        public DateTimeOffset Date { get; set; }
        public DateTimeOffset TimeIn { get; set; }
        public DateTimeOffset? TimeOut { get; set; }
        public AttendanceStatus AttendanceStatus { get; set; }
        public decimal TotalHours { get; set; }
        public Employee? Employee { get; set; }

    }
}

File: Frontend/frontend-emp-system/src/Types/attendance.ts
import type { Employee } from "./employee";
import type { AttendanceStatus } from "./enums";

export interface IAttendanceResponse{
    id:number;
    employeeId: number;
    employeeFirstName : string,
    employeeLastName : string
    date:string;
    timeIn: string;
    timeOut:string;
    attendanceStatus:AttendanceStatus
    totalHours: number;
}

export interface IAttendanceCreateRequest{
    employeeId: number;
    timeIn:string;
}

export interface IAttendanceUpdateRequest{
    timeOut:string;
}

